common/common/segments.py

```diff
@@ -11,6 +11,7 @@
 import os
 import shutil
 import sys
+import warnings
```

Comment:
don't use `warnings`, no-one does outside the standard library. It only ever prints a given warning once, so it's extremely easy to miss. Use `logging.warning()`.